src/Silverback.Integration.Kafka/Messaging/Broker/Kafka/ConsumerChannelsManager.cs

```diff
@@ -267,6 +272,16 @@ await ReadChannelOnceAsync(channelReader, channelIndex, cancellationToken)
             if (consumeResult.IsPartitionEOF)
             {
                 _logger.LogEndOfPartition(consumeResult, _consumer);
+
+                if (_consumer.Endpoint.Configuration.EnablePartitionEof.HasValue &&
```

Comment:
This shouldn't be necessary. With the real thing, it's not possible that you receive the EOF message while `EnablePartitionEof=false`.
This is needed only because the mock doesn't correctly behave.